File: VB/MultiGroupSummary/MyGridControl/MyGridInfoRegistrator.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Grid

Namespace Q354185.MyXtraGrid
    Public Class MyGridInfoRegistrator
        Inherits GridInfoRegistrator

        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New MyGridView(TryCast(grid, GridControl))
        End Function
        Public Overrides Function CreateViewInfo(ByVal view As BaseView) As DevExpress.XtraGrid.Views.Base.ViewInfo.BaseViewInfo
            Return New MyGridViewInfo(CType(view, GridView))
        End Function
        Public Overrides ReadOnly Property ViewName() As String
            Get
                Return "MyGridView"
            End Get
        End Property
        Public Overrides Function CreateHandler(ByVal view As BaseView) As DevExpress.XtraGrid.Views.Base.Handler.BaseViewHandler
            Return New MyGridHandler(TryCast(view, MyGridView))
        End Function
    End Class
End Namespace

File: CS/MultiGroupSummary/MyGridControl/MyGridInfoRegistrator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Grid;

namespace Q354185.MyXtraGrid {
    public class MyGridInfoRegistrator : GridInfoRegistrator {
        public override BaseView CreateView(GridControl grid) {
            return new MyGridView(grid as GridControl);
        }
        public override DevExpress.XtraGrid.Views.Base.ViewInfo.BaseViewInfo CreateViewInfo(BaseView view) {
            return new MyGridViewInfo((GridView)view);
        }
        public override string ViewName {
            get {
                return "MyGridView";
            }
        }
        public override DevExpress.XtraGrid.Views.Base.Handler.BaseViewHandler CreateHandler(BaseView view) {
            return new MyGridHandler(view as MyGridView);
        }
    }
}
